### PR TITLE
User can choose language

### PR DESCRIPTION
**Pull request by ViktorHek**
[User can choose language](https://www.pivotaltracker.com/story/show/176031749)
```
As a user
In order to be able to read articles in my native language,
I would like to switch between languages English and Swedish
```
## Changes proposed in this pull request:
* Visitor can change language to English, swedish or cat-language (a.k.a "Meow")
* Adding Icons that, when clicked, change the language of the page
## What I have learned working on this feature:
* how to change the language for test created in the frontend 
## Packages/Gems added
* i18next
* i18next-xhr-backend
* i18next-browser-languagedetector
## Screenshots (Client)
Screenshot from 2021-01-09 19-25-29

### CODE REVIEW

File: src/i18n.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+import LanguageDetector from "i18next-browser-languagedetector";
+
+i18n
+  .use(Backend)
+  .use(LanguageDetector)
+  .use(initReactI18next)
+  .init({
+    fallbackLng: ["en"],
+    whitelist: ["en", "sv", "cat"],
+    debug: true,
```
Comment (emtalen) [inline src/i18n.js]: you can remove the `debug: true`, it's not needed (sorry for messy demo)

File: src/index.js [hunk 2]
```diff
@@ -8,18 +8,22 @@ import { BrowserRouter } from "react-router-dom";
...
+import "./i18n"
 
 axios.defaults.baseURL = process.env.REACT_APP_API_URL;
 
 const store = configureStore();
 window.store = store;
 
 ReactDOM.render(
-  <Provider store={store}>
+  <Suspense fallback={<div>Loading.... Mauw</div>}>
```
Comment (emtalen) [inline src/index.js]: the `Suspense` should only wrap the `App`

**Review by emtalen — DISMISSED** [review]

File: src/components/Header.jsx [hunk 3]
```diff
@@ -1,11 +1,15 @@
 import React from "react";
 import { NavLink } from "react-router-dom";
-import { Menu, Segment } from "semantic-ui-react";
+import { Menu, Segment, Flag, Icon } from "semantic-ui-react";
 import { useDispatch } from "react-redux";
 import { getArticles } from "../modules/getArticles";
+import i18n from "../i18n";
```
Comment (tochman) [inline src/components/Header.jsx]: I would import this module in `index.js` that is the highest level. Please try that and let me know if you are okay

File: src/index.js (hunk 2 above)
Comment (ViktorHek) [reply src/index.js]: we have moved it inside Provider, but still outside of BrowserRouter

File: src/i18n.js (hunk 1 above)
Comment (ViktorHek) [reply src/i18n.js]: fixed. it's removed :)

File: src/components/Header.jsx (hunk 3 above)
Comment (ViktorHek) [reply src/components/Header.jsx]: we tried but then our tests failed. it said i18n is not defined in the header component

Comment (tochman) [reply src/components/Header.jsx]: I think that is fixable with 
```
const { t, I18n } = useTranslation();
```

Comment (tochman) [reply src/components/Header.jsx]: or perhaps: 
```
const { t, i18n } = useTranslation();
```

Note the lowercased `i` 

Or a combination thereof...

Comment (KyunginNa) [reply src/components/Header.jsx]: The second one worked! Thanks

**Pull request merged**